examples/audio/sound_effects.rs

```diff
@@ -0,0 +1,207 @@
+
+            world.spawn(AudioBundle {
+                source,
+                // We want the sound effect to play once and then despawn.
+                settings,
+            });
+        } else {
+            warn!("Sound effect not found: {}", name);
+        }
+    }
+}
+
+impl FromWorld for SoundEffects {
+    // `FromWorld` is the simplest way to load assets into a resource,
+    // but you likely want to integrate this into your own asset loading strategy.
+    fn from_world(world: &mut World) -> Self {
+        let asset_server = world.get_resource::<AssetServer>().unwrap();
+        let mut map = HashMap::default();
+
+        // Load sound effects here.
+        // Using string parsing to strip numbered suffixes + `AssetServer::load_folder` is a good way to load many sound effects at once, but is not supported on Wasm or Android.
+        let button_press_sfxs = vec![
+            asset_server.load("sounds/button_press_1.ogg"),
+            asset_server.load("sounds/button_press_2.ogg"),
+            asset_server.load("sounds/button_press_3.ogg"),
+        ];
+        map.insert("button_press".to_string(), button_press_sfxs);
+
+        Self { map }
+    }
+}
+
+/// A custom command used to play sound effects.
+struct PlaySoundEffect {
+    name: String,
+    settings: PlaybackSettings,
+}
+
+impl Command for PlaySoundEffect {
+    fn apply(self, world: &mut World) {
+        // Access both the world and the resource we need from it using resource_scope
+        // which temporarily removes the SoundEffects resource from the world
+        world.resource_scope(|world, mut sound_effects: Mut<SoundEffects>| {
+            sound_effects.play(self.name, world, self.settings);
+        });
+    }
+}
+
+/// An "extension trait" used to make it convenient to play sound effects via [`Commands`].
+///
+/// This technique allows us to implement methods for types that we don't own,
+/// which can be used as long as the trait is in scope.
+trait SfxCommands {
+    fn play_sound_effect_with_settings(
+        &mut self,
+        name: impl AsRef<str>,
+        settings: PlaybackSettings,
+    );
+
+    fn play_sound_effect(&mut self, name: impl AsRef<str>) {
+        // This default method implementation saves work for types implementing this trait;
+        // if not overwritten, the trait's default method will be used here, forwarding to the
+        // more customizable method
+        self.play_sound_effect_with_settings(name, PlaybackSettings::DESPAWN);
+    }
+}
+
+impl<'w, 's> SfxCommands for Commands<'w, 's> {
+    // By accepting an `AsRef<str>` here, we can be flexible about what we want to accept:
+    // &str literals are better for prototyping and data-driven sound effects,
+    // but enums are nicer for special-cased effects
+    fn play_sound_effect_with_settings(
+        &mut self,
+        name: impl AsRef<str>,
+        settings: PlaybackSettings,
+    ) {
+        let name = name.as_ref().to_string();
```

Comment:
`as_ref().to_string()` seems like an antipattern. In general converting to owned types should be explicit or be done by the user. Here, it seems like `play_sound_effect_with_settings` wants to borrow `name`, but it actually wants to own it! This should probably accept `Into<String>` instead.